tests/Gedmo/Timestampable/Fixture/Attribute/TitledArticle.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace Gedmo\Tests\Timestampable\Fixture\Attribute;
+
+use Doctrine\DBAL\Types\Types;
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+use Gedmo\Timestampable\Timestampable;
+
+#[ORM\Entity]
+class TitledArticle implements Timestampable
+{
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column(type: Types::INTEGER)]
+    private $id;
+
+    #[ORM\Column(name: 'title', type: Types::STRING, length: 128)]
+    private $title;
+
+    #[ORM\Column(name: 'text', type: Types::STRING, length: 128)]
+    private $text;
+
+    #[ORM\Column(name: 'state', type: Types::STRING, length: 128)]
+    private $state;
+
+    /**
+     * @var \DateTime
+     */
+    #[ORM\Column(name: 'chtext', type: Types::DATETIME_MUTABLE, nullable: true)]
+    #[Gedmo\Timestampable(on: 'change', field: 'text')]
+    private $chText;
+
+    /**
+     * @var \DateTime
+     */
+    #[ORM\Column(name: 'chtitle', type: Types::DATETIME_MUTABLE, nullable: true)]
+    #[Gedmo\Timestampable(on: 'change', field: 'title')]
+    private $chTitle;
+
+    /**
+     * @var \DateTime
+     */
+    #[ORM\Column(name: 'closed', type: TYPES::DATETIME_MUTABLE, nullable: true)]
+    #[Gedmo\Timestampable(on: 'change', field: 'state', value: ['Published', 'Closed'])]
+    private $closed;
+
+    /**
+     * @param \DateTime $chText
+     */
+    public function setChText($chText): void
+    {
+        $this->chText = $chText;
+    }
+
+    /**
+     * @return \DateTime
+     */
+    public function getChText()
+    {
+        return $this->chText;
+    }
+
+    /**
+     * @param \DateTime $chTitle
+     */
+    public function setChTitle($chTitle)
```

Comment:
```suggestion
    public function setChTitle(\DateTime $chTitle): void
```
Please, propagate this change where possible.

Reply:
I was on it already :wink:

Reply:
I also noticed just now that in the change you declared $chTitle as non-nullable, although all the date/time columns actually are nullable. Should I change either the property declarations to non-nullable? I guess it does make sense, since the columns won't be filled initially, but after they are, they shouldn't be turned back to null.